### ISSUE TITLE
Upyutils should use on the device only default functionality

### ISSUE THREAD

**Issue by dsl400** [Carglglz/upydev#33]
It would be nice to be able to use `upydev get main.py` without uploading aditional files onto the device

get the file tree from the device and make all the filtering on the pc  

this concept should be applied to other scripts where possible.

**Comment by Carglglz** [comment]
@dsl400 
> It would be nice to be able to use `upydev get main.py`  without uploading aditional files onto the device

You can, just use `-fg` flag,  see  `upydev get -h`
```
usage:  get [-h] [-dir DIR] [-d D] [-fg] [-b B]
            file/pattern/dir [file/pattern/dir ...]

download files from device

positional arguments:
  file/pattern/dir  indicate a file/pattern/dir to download

optional arguments:
  -h, --help        show this help message and exit
  -dir DIR          path to download from
  -d D              depth level search for pattrn
  -fg               switch off faster get method
  -b B              read buffer for faster get method
```

**Comment by dsl400** [comment]
@Carglglz commented 2 hours ago

2 hours ... about how long it took me to make a PR

**Comment by dsl400** [comment]
#35 will not work because it reaches `self.fastfileio.init_get(dst_name, sz)` 

#35 could be usefull to implement pattern matching when using `-fg `

**Comment by Carglglz** [comment]
> #35 will not work because it reaches `self.fastfileio.init_get(dst_name, sz)` 

True, the size is needed for the progress bar and to check that the file size received matches. So you will need to modify `read_file_tree` to return a list of `(name, size)`  tuples (or `(size, name)` not sure about the order right now)

> #35 could be usefull to implement pattern matching when using `-fg `

Yes that's what I think too 👍🏼

**Comment by dsl400** [comment]
#35 modified 

btw, is there a fast way to upload `nanoglob.py`  ?

**Comment by dsl400** [comment]
it would be nice to present the user with this commands when the scripts are not available instead of some weird error

```
$ upydev get main.py 

nanoglob not availble
use upydev get main.py  -fg or upydev update_upyutils nanoglob.py 

```

**Comment by Carglglz** [comment]
> it would be nice to present the user with this commands when the scripts are not available instead of some weird error

I agree, I should update the docs about `get` command, and improve the error message too. I will do it for next release 👍🏼

**Comment by dsl400** [comment]
can we set arguments like `-fg` in upydev_.config ?

**Comment by Carglglz** [comment]
> can we set arguments like `-fg` in upydev_.config ?

Not right now, however have a look at `register` command, you could register your device and add this flag in the shell function, e.g.
```
$ upydev register -@ esp8266
Registering Device: esp8266 ...
Device: esp8266 registered
Reload .profile with "$ source /Users/$USER/.profile" or open a new terminal to apply the new command
```

and in profile add `-fg` flag in the shell function 
```
#UPYDEV DEVICE esp8266
function esp8266() { upydev "$@" -fg -@ esp8266; }
function _argcomp_upydev() { _python_argcomplete upydev; }
complete -o bashdefault -o default -o nospace -F _argcomp_upydev esp8266
```
Then source `~/.profile` and that's it. 👍🏼

**Comment by dsl400** [comment]
the `register` command scares me because it allows anyone to overwrite system commands without being aware

**Comment by Carglglz** [comment]
> the `register` command scares me because it allows anyone to overwrite system commands without being aware

I think I know what you mean, but it is pretty easy to test if a name is already a command with `which`, `type`, `man`,  or even tab for autocompletion, etc before using `upydev register`, also it can take an `alias` as argument so you can change the name. e.g.

```
$ upydev register esp8 -@ esp8266
Registering Device: esp8266 ...
Device: esp8266 registered as esp8
Reload .profile with "$ source /Users/$USER/.profile" or open a new terminal to apply the new command
```
